### PR TITLE
feat: Add message content intent and additional flags

### PR DESCRIPTION
**Pull request by DenverCoder1**
## Summary

Adds MESSAGE_CONTENT intent and GATEWAY_MESSAGE_CONTENT, GATEWAY_MESSAGE_CONTENT_LIMITED application flags.

In v9, these are given by default, but without this change, in v10, the bot will throw an error if Message Content Intent is disabled in the portal.

This fixes intents to work correctly after migrating to v10

References:

- https://github.com/discord/discord-api-docs/blob/77ee0f906a3d9d2aff0bb3c285913e9db25d98dd/docs/resources/Application.md
- https://github.com/discord/discord-api-docs/blob/3891389606ef8710e7c54d65ae79d0c5313f0247/docs/topics/Gateway.md
- https://github.com/discord/discord-api-docs/discussions/4510
- https://support-dev.discord.com/hc/en-us/articles/4404772028055

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by DenverCoder1** [comment]
**Question:** Should `message_content` be `False` in `Intents.default()`?

Currently that is how it is set so if you want message content in v10, you'd need all() or something like
```py
intents = nextcord.Intents.default()
intents.message_content = True
```

File: nextcord/flags.py [hunk 1]
```diff
@@ -875,6 +876,26 @@ def dm_typing(self):
...
+        - :attr:`Message.content`
+        - :attr:`Message.embeds`
+        - :attr:`Message.attachments`
+        - :attr:`Message.components`
+
+        A bot will always be able to get these attributes from:
+
+        - Messages the bot sends
+        - Messages the bot receives in DMs
+        - Messages in which the bot is mentioned
```
Comment (MaskDuck) [inline nextcord/flags.py:895]: maybe add explict note that you must beg discord for this intents?

Comment (MaskDuck) [reply nextcord/flags.py:895]:
```suggestion
        - Messages in which the bot is mentioned
        .. note::
            Currently, this requires opting in explicitly via the developer portal as well.
            Bots in over 100 guilds will need to apply to Discord for verification.
```

**Review by MaskDuck — CHANGES_REQUESTED** [review]

File: nextcord/flags.py (hunk 1 above)
Comment (MaskDuck) [reply nextcord/flags.py:895]:
```suggestion
        - Messages in which the bot is mentioned
        .. note::
            After April 30, 2022, this requires opting in explicitly via the developer portal as well.
            Bots in over 100 guilds will need to apply to Discord for verification.
```

**Review by MaskDuck — APPROVED** [review]
really gud

File: nextcord/flags.py [hunk 2]
```diff
@@ -884,6 +912,8 @@ def scheduled_events(self):
         - :func:`on_guild_scheduled_event_create`
         - :func:`on_guild_scheduled_event_delete`
         - :func:`on_guild_scheduled_event_update`
+        - :func:`on_guild_scheduled_event_user_add`
```
Comment (TAG-Epic) [inline nextcord/flags.py:915]: Unrelated changes?

**Review by TAG-Epic — CHANGES_REQUESTED** [review]

File: nextcord/flags.py (hunk 2 above)
Comment (DenverCoder1) [reply nextcord/flags.py:915]: It's not related to this change, but I saw it in the [docs](https://github.com/discord/discord-api-docs/blob/3891389606ef8710e7c54d65ae79d0c5313f0247/docs/topics/Gateway.md) when making this change.

Do you want theses lines to be reverted?

Comment (TAG-Epic) [reply nextcord/flags.py:915]: Yeah that wouls be awesome!
